modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractCollocationBenchmark.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.benchmark;
+
+import static org.apache.ignite.catalog.definitions.ColumnDefinition.column;
+import static org.apache.ignite.internal.catalog.CatalogService.DEFAULT_STORAGE_PROFILE;
+import static org.apache.ignite.internal.testframework.TestIgnitionManager.PRODUCTION_CLUSTER_CONFIG_STRING;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.catalog.ColumnType;
+import org.apache.ignite.catalog.definitions.TableDefinition;
+import org.apache.ignite.catalog.definitions.ZoneDefinition;
+import org.apache.ignite.internal.replicator.TablePartitionId;
+import org.apache.ignite.internal.replicator.ZonePartitionId;
+import org.apache.ignite.internal.tx.message.WriteIntentSwitchReplicaRequest;
+import org.apache.ignite.table.KeyValueView;
+import org.apache.ignite.table.Table;
+import org.apache.ignite.table.Tuple;
+
+/**
+ * Base class that allows to measure basic KeyValue operations for tables that share the same distribution zone.
+ * TODO: https://issues.apache.org/jira/browse/IGNITE-22522 remove this benchmark.
+ */
+public class AbstractCollocationBenchmark extends AbstractMultiNodeBenchmark {
+    /** Name of shared zone. */
+    private static final String SHARED_ZONE_NAME = "shared_zone";
+
+    /** System property that allows to enable/disable collocation feature. */
+    private static final String FEATURE_FLAG_NAME = "IGNITE_ZONE_BASED_REPLICATION";
+
+    /** {@code true} enables collocation feature. */
+    private static final Boolean ENABLE_COLLOCATION_FEATURE = true;
+
+    protected final List<KeyValueView<Tuple, Tuple>> tableViews = new ArrayList<>();
+
+    @Override
+    protected int nodes() {
+        return 1;
+    }
+
+    @Override
+    protected int replicaCount() {
+        return 1;
+    }
+
+    @Override
+    protected String clusterConfiguration() {
+        if (tinySchemaSyncWaits()) {
+            return super.clusterConfiguration();
+        } else {
+            // Return a magic string that explicitly requests production defaults.
+            return PRODUCTION_CLUSTER_CONFIG_STRING;
+        }
+    }
+
+    @Override
+    protected void createDistributionZoneOnStartup() {
+        ZoneDefinition zone = ZoneDefinition.builder(SHARED_ZONE_NAME)
+                .partitions(partitionCount())
+                .replicas(replicaCount())
+                .storageProfiles(DEFAULT_STORAGE_PROFILE)
+                .build();
+
+        publicIgnite.catalog().createZone(zone);
+    }
+
+    @Override
+    protected void createTableOnStartup() {
+        for (int i = 1; i <= tableCount(); ++i) {
```

Comment:
The method creates multiple tables, even though its name says that it creates 1 table. How about renaming the method like `createTablesOnStartup()`? This will also cover a case of 1 table.

Reply:
Ok, agree. I will rename the method.